perfkitbenchmarker/providers/gcp/gcp_dpb_dataproc.py

```diff
@@ -0,0 +1,160 @@
+    return stats
+
+  def _Create(self):
+    """Creates the cluster."""
+
+    if self.cluster_id is None:
+      self.cluster_id = 'pkb-' + FLAGS.run_uri
+    cmd = util.GcloudCommand(self, 'dataproc', 'clusters', 'create',
+                             self.cluster_id)
+    if self.project is not None:
+      cmd.flags['project'] = self.project
+
+    # The number of worker machines in the cluster
+    cmd.flags['num-workers'] = self.spec.worker_count
+
+    # Initialize applications on the dataproc cluster
+    if self.spec.applications:
+      logging.info('Include the requested applications')
+
+    for role in ['worker', 'master']:
+      # Set machine type
+      if self.spec.worker_group.vm_spec.machine_type:
+        self.add_to_cmd(cmd, '{0}-machine-type'.format(role),
+                        self.spec.worker_group.vm_spec.machine_type)
+
+      # Set boot_disk_size
+      if self.spec.worker_group.vm_spec.boot_disk_size:
+        self.add_to_cmd(cmd, '{0}-boot-disk-size'.format(role),
+                        self.spec.worker_group.vm_spec.boot_disk_size)
+
+      # Set ssd count
+      if self.spec.worker_group.vm_spec.num_local_ssds:
+        self.add_to_cmd(cmd, 'num-{0}-local-ssds'.format(role),
+                        self.spec.worker_group.vm_spec.num_local_ssds)
+    # TODO(saksena): Retrieve the cluster create time and hold in a var
+    cmd.Issue()
+
+  def _Delete(self):
+    """Deletes the cluster."""
+    cmd = util.GcloudCommand(self, 'dataproc', 'clusters', 'delete',
+                             self.cluster_id)
+    cmd.Issue()
+
+  def _Exists(self):
+    """Check to see whether the cluster exists."""
+    cmd = util.GcloudCommand(self, 'dataproc', 'clusters', 'describe',
+                             self.cluster_id)
+    _, _, retcode = cmd.Issue()
+    return retcode == 0
+
+
+  def submit_job(self, jarfile, classname, job_poll_interval=None,
+                 job_arguments=None, job_stdout_file=None,
+                 job_type=None):
+    cmd = util.GcloudCommand(self, 'dataproc', 'jobs', 'submit', job_type)
+    cmd.flags['cluster'] = self.cluster_id
+
+    if classname:
+      cmd.flags['jars'] = jarfile
+      cmd.flags['class'] = classname
+    else:
+      cmd.flags['jar'] = jarfile
+
+    # Dataproc gives as stdout an object describing job execution.
+    # Its stderr contains a mix of the stderr of the job, and the
+    # stdout of the job.  We set the driver log level to FATAL
+    # to suppress those messages, and we can then separate, hopefully
+    # the job standard out from the log messages.
+    cmd.flags['driver-log-levels'] = 'root=FATAL'
+
+    if job_arguments:
+      cmd.additional_flags = ['--'] + job_arguments
+
+    stdout, stderr, retcode = cmd.Issue(timeout=None)
+    if retcode != 0:
+      return {dpb_service.SUCCESS: False}
+
+    stats = self._GetStats(stdout)
+    return stats
+
+  def SetClusterProperty(self):
+    pass
+
+  def add_to_cmd(self, cmd, cmd_property, cmd_value):
```

Comment:
Changed the name to better match the existing methods and put an '_' to mark it as internal